packages/web3-core-method/src/index.js

```diff
@@ -303,7 +310,7 @@ Method.prototype._confirmTransaction = function (defer, result, payload) {
                             promiseResolved = true;
                         }
 
-                        utils._fireError(new Error('The transaction receipt didn\'t contain a contract address.'), defer.eventEmitter, defer.reject);
+                        utils._fireError(new Error('The transaction receipt didn\'t contain a contract address.'), defer.eventEmitter, defer.reject, receipt);
```

Comment:
I like the 'list' arg style you've done below...it's readable within the github page :)

It looks like `_fireError` has five params - the fourth is a callback, the fifth is optionalData?
```suggestion
                        utils._fireError(
                            new Error('The transaction receipt didn\'t contain a contract address.'), 
                            defer.eventEmitter, 
                            defer.reject, 
                            null,
                            receipt
                        );
```